src/Lightweight/SqlStatement.hpp

```diff
@@ -321,6 +321,67 @@ class [[nodiscard]] SqlResultCursor
     SqlStatement* m_stmt;
 };
 
+// input iterator
+template <typename T>
+class iterator
+{
+  public:
+    using difference_type = bool;
+    using value_type = T;
+
+    iterator(SqlStatement& stmt, std::string_view table_name):
+        m_stmt(&stmt),
+        m_table_name(table_name)
```

Comment:
```suggestion
        m_table_name(RecordTableName<T>)
```
If `T` is supposed to be the record struct, then we do not need to pass this string_view to the constructor.